modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java

```diff
@@ -307,55 +285,47 @@ public void start(NHttpClientConnection conn) throws IOException, HttpException
-							MessageFormatter formatter =  MessageProcessorSelector.getMessageFormatter(requestMsgCtx);
-							OMOutputFormat format = PassThroughTransportUtils.getOMOutputFormat(requestMsgCtx);
-							formatter.writeTo(requestMsgCtx, format, out, false);
-							OutputStream _out = pipe.getOutputStream();
-							IOUtils.write(out.toByteArray(), _out);
-						
-							entity.setContentLength(new Long(out.toByteArray().length));
-							entity.setChunked(false);
-						}
-					}
-					// pipe.setSerializationComplete(true);
-				} catch (XMLStreamException e) {
-					 e.printStackTrace();
-				
-				}
-			}
-
-		}  
+    /**
+     * Handles the chuking messages in Passthough context, create a temporary buffer and calculate the message
+     * size before writing to the external buffer, which is required the context of handling DISABLED chunking
+     * messages
+     *
+     * @param conn
+     * @param requestMsgCtx
+     * @throws IOException
+     * @throws AxisFault
+     */
+    private void processChunking(NHttpClientConnection conn, MessageContext requestMsgCtx) throws IOException, AxisFault {
+        String disableChunking = (String) requestMsgCtx.getProperty(PassThroughConstants.DISABLE_CHUNKING);
+        String forceHttp10 = (String) requestMsgCtx.getProperty(PassThroughConstants.FORCE_HTTP_1_0);
+        if ("true".equals(disableChunking) || "true".equals(forceHttp10)) {
+            if (requestMsgCtx.getEnvelope().getBody().getFirstElement() == null) {
+                BasicHttpEntity entity = (BasicHttpEntity) ((BasicHttpEntityEnclosingRequest) request).getEntity();
+                try {
+                    RelayUtils.buildMessage(requestMsgCtx);
+                    this.hasEntityBody = true;
+                    Pipe pipe = (Pipe) requestMsgCtx.getProperty(PassThroughConstants.PASS_THROUGH_PIPE);
+                    if (pipe != null) {
+                        pipe.attachConsumer(conn);
+                        this.connect(pipe);
+                        if (Boolean.TRUE.equals(requestMsgCtx.getProperty(PassThroughConstants.MESSAGE_BUILDER_INVOKED))) {
+                            ByteArrayOutputStream out = new ByteArrayOutputStream();
+                            MessageFormatter formatter =  MessageProcessorSelector.getMessageFormatter(requestMsgCtx);
+                            OMOutputFormat format = PassThroughTransportUtils.getOMOutputFormat(requestMsgCtx);
+                            formatter.writeTo(requestMsgCtx, format, out, false);
+                            OutputStream _out = pipe.getOutputStream();
+                            IOUtils.write(out.toByteArray(), _out);
+                            entity.setContentLength(new Long(out.toByteArray().length));
+                            entity.setChunked(false);
+                        }
+                    }
+                    // pipe.setSerializationComplete(true);
+                } catch (XMLStreamException e) {
+                     e.printStackTrace();
```

Comment:
Use a logger instead of printStackTrace.

Reply:
I changed to use logger as you mentioned.